### PR TITLE
Passed user chat data between pages, added text parser, made the chat/message cards smart widgets

### PR DESCRIPTION
**Pull request by pauleke65**

### CODE REVIEW

File: lib/ui/shared/smart_widgets/text_parser/text_parser_viewmodel.dart [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        }),
+    MatchText(
+      pattern: r"\[(@[^:]+):([^\]]+)\]",
+      style: AppTextStyles.messageTextButton,
+
+      renderText: ({required pattern, required str}) {
+        RegExp customRegExp = RegExp(r"\[(@[^:]+):([^\]]+)\]");
+        Match match = customRegExp.firstMatch(str)!;
+
+        print('test test: ${match[1]}');
```
Comment (isaacadariku) [inline lib/ui/shared/smart_widgets/text_parser/text_parser_viewmodel.dart:56]: remove all print statement and the commented codes

File: lib/ui/shared/smart_widgets/thread_card/widgets/thread_card_main.dart [hunk 2]
```diff
@@ -64,17 +66,28 @@ class ThreadCardMain extends ViewModelWidget<ThreadCardViewModel> {
...
-                EmojisList(userPost: userPost),
-                SizedBox(height: 15)
-              ],
-            ),
-          )
-        ],
+                  SizedBox(height: 10),
+                  EmojisList(userPost: userPost),
+                  SizedBox(height: 15),
+                  userPost!.userThreadPosts != null
```
Comment (isaacadariku) [inline lib/ui/shared/smart_widgets/thread_card/widgets/thread_card_main.dart:77]: I just notice this logic is used in the view, it should not, the value should be gotten from the viewmodel

File: lib/ui/shared/styles.dart [hunk 3]
```diff
@@ -50,6 +50,25 @@ class AppTextStyles {
     fontWeight: FontWeight.w400,
     fontSize: 12,
   );
+  static TextStyle messageText = GoogleFonts.lato(
```
Comment (isaacadariku) [inline lib/ui/shared/styles.dart:53]: Were this styles added to the design or just for the UI's you built ?

File: lib/ui/view/threads/all_threads/threads_viewmodel.dart [hunk 4]
```diff
@@ -12,18 +12,17 @@ class ThreadsViewModel extends BaseViewModel {
   final _navigationService = locator<NavigationService>();
 
   List<UserPost> userPosts = userPost;
-  
 
   void initialise() {
     //refresh page
+    print('Hello there');
```
Comment (isaacadariku) [inline lib/ui/view/threads/all_threads/threads_viewmodel.dart:18]: Please use a logger and ensure your code is clean by removing all print statement with commented codes

File: lib/ui/view/threads/all_threads/threads_viewmodel.dart [hunk 5]
```diff
@@ -33,7 +32,8 @@ class ThreadsViewModel extends BaseViewModel {
     _navigationService.back();
   }
 
-  Future navigateToThread() async {
-    _navigationService.navigateTo(Routes.threadDetailView);
+  Future navigateToThread(UserPost? userPost) async {
```
Comment (isaacadariku) [inline lib/ui/view/threads/all_threads/threads_viewmodel.dart:35]: Can the future have a type of void or change the method to a void type

**Review by isaacadariku — CHANGES_REQUESTED** [review]
Great Paul 😄 Please can we ensure we remove every  `userPost` logic on the UI and some other few changes requested then we are good to go. Weldone

File: lib/ui/shared/styles.dart (hunk 3 above)
Comment (pauleke65) [reply lib/ui/shared/styles.dart:53]: I added this for the UI I built, to serve as the text style for all the messages handled by the Thread cards.

**Review by pauleke65 — COMMENTED** [review]
All requested changes made

**Review by isaacadariku — APPROVED** [review]
LGTM

**Review by mariamadebolahamzat — APPROVED** [review]

**Comment by Blazebrain** [comment]
Fix conflicts

**Review by Blazebrain — APPROVED** [review]

**Review by mariamadebolahamzat — APPROVED** [review]

**Pull request merged**